pkg/apis/ceph.rook.io/s3ext/api.go

```diff
@@ -0,0 +1,88 @@
+package s3ext
+
+import (
+	"github.com/aws/aws-sdk-go/aws/awsutil"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+type DeleteBucketNotificationRequestInput struct {
+	_ struct{} `locationName:"DeleteBucketNotificationRequestInput" type:"structure"`
+
+	// The name of the bucket for which to get the notification configuration.
+	//
+	// Bucket is a required field
+	Bucket *string `location:"uri" locationName:"Bucket" type:"string" required:"true"`
```

Comment:
Why is it a pointer if it's required?

Reply:
i copy&pasted from the AWS SDK

Reply:
I wonder what the purpose is... the only reason I can think to do this is if we want to support the empty string `""` as a valid value, in which case we would need the `nil` case.

Reply:
this is an extension to the AWS SDK. they do not support deleting bucket notifications